pallets/aggregate/src/lib.rs

```diff
@@ -630,20 +750,44 @@ pub mod pallet {
                 aggregation_id,
                 receipt: root,
             });
+
+            let domain = Domains::<T>::get(domain_id).ok_or(Error::<T>::UnknownDomainId)?;
+            let dispatch_config = &domain.dispatch_config;
+
+            let destination = Destination::Hyperbridge(HyperBridgeDispatchParameters {
+                aggregation_id,
+                aggregation: root,
+                dispatch_config: DispatchConfig {
+                    destination_chain: dispatch_config.destination_chain.clone(),
+                    destination_module: dispatch_config.destination_module,
+                    timeout: dispatch_config.timeout,
+                    base_fee: dispatch_config.base_fee,
+                    gas_price: dispatch_config.gas_price,
+                },
+            });
```

Comment:
In my opinion, what we need to save in this pallet are destination_chain, destination_module, timeout, base_fee and gas_price. This should be wrapped in a Destination enum.
On aggregate receives (domain_id, aggregation_id, aggregation, destination) and the implementer of it (e.g. router) internally does the processing to send correctly formatted stuff to hyperbridge pallet.

Reply:
I see, I think you are saying Destination should be dispatch_config and then this part should be implemented in the on_aggregate implementation in the runtime